### PR TITLE
Refrigeration Components Plant Component Refactor

### PR DESCRIPTION
**Pull request by mitchute**
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mitchute** [comment]
@Myoldmopar I addressed the bug causing the single-file failure from the work yesterday. If CI comes back OK, I believe this one is ready.

**Comment by Myoldmopar** [comment]
CI is very happy with this, I'll review now.

File: src/EnergyPlus/RefrigeratedCase.hh [hunk 1]
```diff
@@ -55,260 +55,20 @@
 // EnergyPlus Headers
 #include <EnergyPlus/DataGlobals.hh>
 #include <EnergyPlus/EnergyPlus.hh>
+#include <EnergyPlus/PlantComponent.hh>
 
 namespace EnergyPlus {
 
 namespace RefrigeratedCase {
 
-    // Using/Aliasing
```
Comment (Myoldmopar) [inline src/EnergyPlus/RefrigeratedCase.hh:63]: So these were all in the header but none were needed.  That is awesome.  Good cleanup!

File: src/EnergyPlus/PlantLoopEquip.cc [hunk 2]
```diff
@@ -911,18 +910,10 @@ namespace PlantLoopEquip {
...
-                }
+                sim_component.compPtr->simulate(sim_component_location, FirstHVACIteration, CurLoad, RunFlag);
 
             } else if (EquipTypeNum == TypeOf_RefrigerationWaterCoolRack) {
-                SimRefrigCondenser(EquipTypeNum, sim_component.Name, EquipNum, FirstHVACIteration, InitLoopEquip);
-
-                if (InitLoopEquip) {
-                    sim_component.CompNum = EquipNum;
-                }
+                sim_component.compPtr->simulate(sim_component_location, FirstHVACIteration, CurLoad, RunFlag);
```
Comment (Myoldmopar) [inline src/EnergyPlus/PlantLoopEquip.cc:916]: Makes good sense :wink:

File: src/EnergyPlus/Plant/PlantManager.cc [hunk 3]
```diff
@@ -1433,10 +1434,12 @@ namespace EnergyPlus {
                                 this_comp.TypeOf_Num = TypeOf_RefrigSystemWaterCondenser;
                                 this_comp.GeneralEquipType = GenEquipTypes_Refrigeration;
                                 this_comp.CurOpSchemeType = DemandOpSchemeType;
+                                this_comp.compPtr = RefrigeratedCase::RefrigCondenserData::factory(CompNames(CompNum));
```
Comment (Myoldmopar) [inline src/EnergyPlus/Plant/PlantManager.cc:1437]: Standard, good.

File: tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.cc [hunk 4]
```diff
@@ -444,6 +445,7 @@ void EnergyPlusFixture::clear_all_states()
     Pumps::clear_state();
     PurchasedAirManager::clear_state();
     PVWatts::clear_state();
+    RefrigeratedCase::clear_state();
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.cc:448]: And another `clear_state`; great!

**Review by Myoldmopar — APPROVED** [review]
No problems here, this is good stuff.

**Comment by Myoldmopar** [comment]
Thanks @mitchute; merging this now.

**Pull request merged**